docs/adr/0016-use-graphql-and-apollo-for-settings-redesign.md

```diff
@@ -0,0 +1,136 @@
+# Use GraphQL and Apollo for Settings Redesign
+
+- Deciders: Lauren Zugai, Ben Bangert, Wil Clouser, Jody Heavener, Les Orchard, Danny Coates, Barry Chen, Vijay Budhram, Dave Justice, Alex Davis
+- Date: 2020-05-05
+
+## Context and Problem Statement
+
+The [Settings Redesign project](https://jira.mozilla.com/browse/FXA-840) will be created [as a new React application](https://github.com/mozilla/fxa/blob/master/docs/adr/0011-create-new-react-app-for-settings-redesign.md) and in turn, has opened the door to assess certain pieces of our technology stack.
+
+[GraphQL](https://graphql.org/), or GQL, is not a database query language, but is instead a query language for APIs. It _describes_ data requirements, and is a powerful alternative to REST. Some benefits can be gained by using it on top of existing REST architecture.
+
+"Apollo" in this document refers to the "Apollo client" and "Apollo server" pieces of the Apollo platform¹, which can be described as a unified data layer that enables applications to interact with data from data stores and APIs. In other words, it allows us to write and handle GraphQL on the client and server. Apollo also gives us many tools out of the box like caching.
+
+This ADR serves to lay out pros and cons of using GraphQL and Apollo in the Settings Redesign project as an alternative to hitting our conventional REST endpoints.
+
+¹Apollo also offers Apollo Graph Manger and Apollo Federation which are paid services, [read more from their docs](https://www.apollographql.com/docs/intro/platform/). We do not need to use these to use GQL with Apollo server or Apollo client.
+
+## Decision Drivers
+
+- Performance
+- Easy API setup and React integration
+- Clarity around "what's happening?" and data received from the server
+- Development speed, both around initial setup and as new features roll out
+
+## Considered Options
+
+- Option A - Send standard CRUD requests to our REST endpoints
```

Comment:
This is a bit unfair to REST, but unfortunately we all know what you mean by that.

Reply:
I may add a couple more pros to option A for just REST in general. I definitely spent more time on the GQL pros section.

Reply:
Oh, I don't know if it's intentional, but that seems to equate REST with CRUD.  While REST is not CRUD, as suggested my in previous comment, unfortunately that's what people think anyways.  I was not suggesting any changes. 😃